mate-panel/libpanel-util/panel-session-manager.c

```diff
@@ -42,136 +49,102 @@ panel_session_manager_class_init (PanelSessionManagerClass *klass)
 
 	object_class = G_OBJECT_CLASS (klass);
 
-	object_class->constructor = panel_session_manager_constructor;
+	object_class->finalize = panel_session_manager_finalize;
 }
 
 static void
 panel_session_manager_init (PanelSessionManager *manager)
 {
-}
-
-static GObject *
-panel_session_manager_constructor (GType                  type,
-				   guint                  n_construct_properties,
-				   GObjectConstructParam *construct_properties)
-{
-	GObject *obj;
-	GError  *error;
-
-	obj = G_OBJECT_CLASS (panel_session_manager_parent_class)->constructor (
-							type,
-							n_construct_properties,
-							construct_properties);
-
-
-	panel_dbus_service_define_service (PANEL_DBUS_SERVICE (obj),
-					   "org.gnome.SessionManager",
-					   "/org/gnome/SessionManager",
-					   "org.gnome.SessionManager");
-
-	error = NULL;
-	if (!panel_dbus_service_ensure_connection (PANEL_DBUS_SERVICE (obj),
-						   &error)) {
-		g_message ("Could not connect to session manager: %s",
-			   error->message);
+	GError *error = NULL;
+
+	manager->proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
```

Comment:
Would not be better to move this to `ensure_proxy` or something like that? And use it in all functions that use proxy? Not sure if that can happen, but if proxy creating fails then you will never try again to create proxy. Old code would try to create proxy if it is `NULL`...